lib/DeployActions/DockerActions.php

```diff
@@ -75,18 +80,176 @@ public function deployExApp(ExApp $exApp, DaemonConfig $daemonConfig, array $par
 			return $result['error'];
 		}
 		$this->exAppService->setAppDeployProgress($exApp, 97);
-		$result = $this->startContainer($dockerUrl, $this->buildExAppContainerName($params['container_params']['name']));
+
+		$this->updateCerts($dockerUrl, $containerName);
+		$this->exAppService->setAppDeployProgress($exApp, 98);
+
+		$result = $this->startContainer($dockerUrl, $containerName);
 		if (isset($result['error'])) {
 			return $result['error'];
 		}
 		$this->exAppService->setAppDeployProgress($exApp, 99);
-		if (!$this->waitTillContainerStart($this->buildExAppContainerName($exApp->getAppid()), $daemonConfig)) {
+		if (!$this->waitTillContainerStart($containerName, $daemonConfig)) {
 			return 'container startup failed';
 		}
 		$this->exAppService->setAppDeployProgress($exApp, 100);
 		return '';
 	}
 
+	private function updateCerts(string $dockerUrl, string $containerName): void {
+		try {
+			$this->startContainer($dockerUrl, $containerName);
+
+			$osInfo = $this->getContainerOsInfo($dockerUrl, $containerName);
+			if (!$this->isSupportedOs($osInfo)) {
+				$this->logger->warning(sprintf(
+					"Unsupported OS detected for container: %s. OS info: %s",
+					$containerName,
+					$osInfo
+				));
+				return;
+			}
+
+			$bundlePath = $this->certificateManager->getAbsoluteBundlePath();
+			$targetDir = $this->getTargetCertDir($osInfo); // Determine target directory based on OS
+			$this->executeCommandInContainer($dockerUrl, $containerName, ['mkdir', '-p', $targetDir]);
+			$this->installParsedCertificates($dockerUrl, $containerName, $bundlePath, $targetDir);
+
+			$updateCommand = $this->getCertificateUpdateCommand($osInfo);
+			$this->executeCommandInContainer($dockerUrl, $containerName, $updateCommand);
+		} catch (Exception $e) {
+			$this->logger->warning(sprintf(
+				"Failed to update certificates in container: %s. Error: %s",
+				$containerName,
+				$e->getMessage()
+			));
+		} finally {
+			$this->stopContainer($dockerUrl, $containerName);
+		}
+	}
+
+	private function parseCertificatesFromBundle(string $bundlePath): array {
+		$contents = file_get_contents($bundlePath);
+
+		// Match only certificates
+		preg_match_all('/-----BEGIN CERTIFICATE-----(.+?)-----END CERTIFICATE-----/s', $contents, $matches);
+
+		return $matches[0] ?? [];
+	}
+
+	private function installParsedCertificates(string $dockerUrl, string $containerId, string $bundlePath, string $targetDir): void {
+		$certificates = $this->parseCertificatesFromBundle($bundlePath);
+		$tempDir = sys_get_temp_dir();
+
+		foreach ($certificates as $index => $certificate) {
+			$tempFile = $tempDir . "/{$containerId}_cert_{$index}.crt";
+			if (file_exists($tempFile)) {
+				unlink($tempFile);
+			}
+			file_put_contents($tempFile, $certificate);
+
+			// Build the path in the container
+			$pathInContainer = $targetDir . "/custom_cert_$index.crt";
```

Comment:
We don't actually have file names, because all certificates are often in one file in a heap.

To save each certificate by name, we need to use openssl to process them, get their name and then we can do this.

We won't complicate things like that for now, first let's make sure that this option works.

Reply:
sure, makes sense. I suggested that assuming the names were already there.